### DISCUSSION TITLE
Running archivebox docker without the django server

### DISCUSSION THREAD

**Discussion by mtvu** [ArchiveBox/ArchiveBox#688] [Q&A] [answered]
Hi all, 

i'm trying to run archivebox docker  on a VPS without exposing the public webpage. I know that I could go with the option of using service like Cloudflare or putting archivebox behind a reverse proxy with auth on. But I just want to have a simpler option where I could have archivebox running without the django server. Does anyone try this before ?

Thanks,

**Comment by pirate** [comment]
This was actually the primary way that archivebox was used in the past. `./archive ...` produced a static `index.html` with no server and you could serve it however you wanted.
 It was changed over time to preferring `archivebox server` because it's really dangerous to serve raw archived html/js from your filesystem without some headers and restrictions.
 
 If you want the static behavior, just make sure you serve it on a completely separate domain from every other domain you use. This is why github uses `githubusercontent.com` and google uses `googleusercontent.com` for everything user-uploaded, it's very dangerous to serve content from a domain shared with others. (think about cookies, auth tokens, and all manner of other CORS dangerousness that happens when any page archived can run arbitrary JS when you view it)
 
 ```bash
 archivebox list --html --with-headers > index.html
 
 open index.html
 python3 -m http.server --bind 0.0.0.0
 ...
 ```
 ```

**Reply by mtvu** [reply]
Thanks for the tip.  I was also looking for a way to stop django server running inside docker. ps is not available so I cannot just pkill the process. I think the server is turned on by default.

**Reply by pirate** [reply]
There is no need to start and then kill the server, just pass it a different startup cmd in the `command:` section or by using CLI args if you don't want it to run the server.

```yaml
services:
    archivebox:
        command: archivebox update  # or any other command here
        ...
```
Or
```bash
docker run -v $PWD:/data archivebox/archivebox add 'https://...'
docker run -v $PWD:/data archivebox/archivebox update
docker run -it -v $PWD:/data archivebox/archivebox list --html --with-headers > index.html
...

docker run -v $PWD:/data archivebox/archivebox help
```

**Comment by mtvu** [comment, marked as answer]
Thanks @pirate. I've got it working today
